app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+                text = name,
+                style = MaterialTheme.typography.h5,
+                modifier = Modifier.padding(start = 8.dp, end = 8.dp, bottom = 16.dp)
+                    .gravity(Alignment.CenterHorizontally).onPositioned {
+                        onNamePositioned(it)
+                    }.plus(hideModifier)
+            )
+        }
+        Text(
+            text = stringResource(id = R.string.watering_needs_prefix),
+            color = MaterialTheme.colors.primaryVariant,
+            style = MaterialTheme.typography.body1.copy(
+                fontWeight = FontWeight.Bold
+            ),
+            modifier = Modifier.padding(horizontal = 8.dp).gravity(Alignment.CenterHorizontally)
+        )
+        Text(
+            text = getQuantityString(R.plurals.watering_needs_suffix, wateringInterval),
+            color = MaterialTheme.colors.onBackground.copy(alpha = 0.6f),
+            modifier = Modifier.padding(start = 8.dp, end = 8.dp, bottom = 16.dp)
+                .gravity(Alignment.CenterHorizontally)
+        )
+        PlantDescription(description)
+    }
+}
+
+@Composable
+private fun PlantDescription(description: String) {
+    // Issue: User input doesn't work properly, HTML links cannot be clicked - b/158088138
+    // HTML support coming to Compose - b/139320905
+    AndroidView(resId = R.layout.item_plant_description) {
+        (it as TextView).text = HtmlCompat.fromHtml(description, HtmlCompat.FROM_HTML_MODE_COMPACT)
+    }
+}
+
+/**
+ * Gives a parallax effect to the content at the top of a [VerticalScroller].
+ */
+@Composable
+private fun ParallaxEffect(
+    scrollerPosition: ScrollerPosition,
+    modifier: Modifier = Modifier,
+    content: @Composable (Modifier) -> Unit
+) {
+    val offset = scrollerPosition.value / ParallaxDelta
+    val offsetDp = with(DensityAmbient.current) { offset.toDp() }
+    content(Modifier.padding(top = offsetDp).plus(modifier))
+}
+
+/**
+ * Calculates offset FAB needs to keep aligned in the middle of the bottom of the picture.
+ *
+ * As the [Modifier.onPositioned] in the image is invoked after scrollPosition has changed,
+ * there's a frame delay.
+ */
+@Composable
+private fun getFabOffset(imageHeight: Int, scrollerPosition: ScrollerPosition): Dp {
+    return with(DensityAmbient.current) {
+        imageHeight.toDp() + (scrollerPosition.value / ParallaxDelta).toDp() - (56 / 2).dp
+    }
+}
+
+/**
+ * Hides an element on the screen leaving its space occupied.
+ * This should be replaced with the visible modifier in the future: b/158837937
+ *
+ * Disclaimer: This implementation assumes that the content is visible before hiding it.
+ */
+@Composable
+private fun Hide(hide: Boolean, content: @Composable (Modifier) -> Unit) {
+    var contentSize by state { IntSize.Zero }
+    if (hide) {
+        val (width, height) = remember(contentSize) {
+            with(DensityAmbient.current) {
+                contentSize.width.toDp() to contentSize.height.toDp()
```

Comment:
This conversion is annoying. Worth a FR for some `preferredPxSize`?

Reply:
Probably? Although they'll tell me to create a custom layout to do this haha I should probably give it a go

Reply:
Created a modifier instead! Thanks